### ISSUE TITLE
Update Of Object Returning Old Link Set

### ISSUE THREAD

**Issue by LukeLaScala** [edgedb/edgedb#6523]
- EdgeDB Version: `4.0+17135b8`
- EdgeDB CLI Version: `4.0.2+500be79`
- OS Version: `MacOS Venture 13.3.1 (a)`

Here is the offending query: https://pastebin.com/raw/A6DSbYZJ. 

Relevant Schema: https://pastebin.com/raw/gxPYbwyi

Issue: When adding a `booking` to an `availability` and then subsequently (in the same query) selecting that `availability`, I'm not seeing links of the `booking` object populated correctly. The `selections` inside the new booking object that I'm adding are showing as empty sets in the returned object, but they are in fact being populated.

In short, when updating a link property in a deeply nested object using with, I'm not seeing the new value of this link.
